src/utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module horizontal-rule/utils
+ */
+
+import { findOptimalInsertionPosition, isWidget, toWidget } from '@ckeditor/ckeditor5-widget/src/utils';
+
+/**
+ * Converts a given {@link module:engine/view/element~Element} to a horizontal rule widget:
+ * * Adds a {@link module:engine/view/element~Element#_setCustomProperty custom property} allowing to
+ * recognize the horizontal rule widget element.
+ * * Calls the {@link module:widget/utils~toWidget} function with the proper element's label creator.
+ *
+ * @param {module:engine/view/element~Element} viewElement
+ * @param {module:engine/view/downcastwriter~DowncastWriter} writer An instance of the view writer.
+ * @param {String} label The element's label.
+ * @returns {module:engine/view/element~Element}
+ */
+export function toHorizontalRuleWidget( viewElement, writer, label ) {
```

Comment:
It's used in a single place in the plugin, so there's no point to having it defined outside.